internal/alpini/netty4/alpini-netty4-base/src/main/java/com/linkedin/alpini/netty4/handlers/ConnectionControlHandler.java

```diff
@@ -32,27 +34,31 @@
  */
 @ChannelHandler.Sharable
 public class ConnectionControlHandler extends ConnectionLimitHandler {
+  private static final Logger LOGGER = LogManager.getLogger(ConnectionControlHandler.class);
   private final LongAdder _activeCount = new LongAdder();
   private final LongAdder _inactiveCount = new LongAdder();
```

Comment:
Just a question about why using `LongAddr` here? I think `AtomicInt` or `AtomicLong` is quite convenient and this handler is shared, so at most we only need to create one atomic variable.

Not a blocker.

Reply:
These are implementations from Antony, previous Alpini owner. It appears that LongAdder performs better than AtomicLong during high contention on updating the value because LongAdder would maintain multiple counters and only merge then during read.